include/ocpp/common/evse_security.hpp

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2020 - 2023 Pionix GmbH and Contributors to EVerest
+#ifndef OCPP_COMMON_EVSE_SECURITY
+#define OCPP_COMMON_EVSE_SECURITY
+
+#include <filesystem>
+#include <fstream>
+#include <memory>
+#include <string>
+
+#include <ocpp/common/types.hpp>
+
+namespace ocpp {
+
+/// \brief Handler for security related operations of the charging station
+class EvseSecurity {
+
+public:
+    /// \brief Installs the CA \p certificate for the given \p certificate_type . This function respects the
+    /// requirements of OCPP specified for the CSMS initiated message InstallCertificate.req .
+    /// \param certificate PEM formatted CA certificate
+    /// \param certificate_type specifies the CA certificate type
+    /// \return result of the operation
+    virtual InstallCertificateResult install_ca_certificate(const std::string& certificate,
+                                                            const CaCertificateType& certificate_type) = 0;
```

Comment:
I was wondering this in our call earlier today too. Why are we passing enum class arguments by const reference? Seems like by value would be simpler and more code efficient as well.

Reply:
this is just a matter of habbit, they could also be passed by value. In general, there are a couple of things we could target for a larger styling refactor of libocpp, which is already on our list :)